lib/srv/discovery/fetchers/db/aws_elasticache_serverless.go

```diff
@@ -0,0 +1,132 @@
+func (f *elastiCacheServerlessPlugin) GetDatabases(ctx context.Context, cfg *awsFetcherConfig) (types.Databases, error) {
+	awsCfg, err := cfg.AWSConfigProvider.GetConfig(ctx, cfg.Region,
+		awsconfig.WithAssumeRole(cfg.AssumeRole.RoleARN, cfg.AssumeRole.ExternalID),
+		awsconfig.WithCredentialsMaybeIntegration(awsconfig.IntegrationMetadata{Name: cfg.Integration}),
+	)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	clt := cfg.awsClients.GetElastiCacheClient(awsCfg)
+	caches, err := getElastiCacheServerlessCaches(ctx, clt)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	var eligibleCaches []ectypes.ServerlessCache
+	for _, cache := range caches {
+		if !libcloudaws.IsElastiCacheServerlessCacheSupported(&cache) {
+			cfg.Logger.DebugContext(ctx, "Skipping unsupported ElastiCache cluster",
+				"cluster", aws.ToString(cache.ServerlessCacheName),
+			)
+			continue
+		}
+		if !libcloudaws.IsElastiCacheServerlessCacheAvailable(&cache) {
+			cfg.Logger.DebugContext(ctx, "Skipping unavailable ElastiCache Serverless cache",
+				"cache_name", aws.ToString(cache.ServerlessCacheName),
+				"status", aws.ToString(cache.Status),
+			)
+			continue
+		}
+		eligibleCaches = append(eligibleCaches, cache)
+	}
+
+	if len(eligibleCaches) == 0 {
+		return nil, nil
+	}
+
+	var databases types.Databases
+	for _, cache := range eligibleCaches {
+		// Resource tags are not found in ectypes.ServerlessCache but can
+		// be on obtained by ectypes.ListTagsForResource (one call per
+		// resource).
+		tags, err := getElastiCacheResourceTags(ctx, clt, cache.ARN)
+		if err != nil {
+			if trace.IsAccessDenied(err) {
+				cfg.Logger.DebugContext(ctx, "No permissions to list resource tags", "error", err)
+			} else {
+				cfg.Logger.InfoContext(ctx, "Failed to list resource tags for ElastiCache Serverless cache",
+					"cache_name", aws.ToString(cache.ServerlessCacheName),
+					"error", err,
+				)
+			}
+		}
+
+		extraLabels := common.ExtraElastiCacheServerlessLabels(&cache, tags)
+		if db, err := common.NewDatabaseFromElastiCacheServerlessCache(&cache, extraLabels); err != nil {
+			cfg.Logger.InfoContext(ctx, "Could not convert ElastiCache Serverless cache to database resource",
+				"cache_name", aws.ToString(cache.ServerlessCacheName),
+				"error", err,
+			)
+		} else {
+			databases = append(databases, db)
+		}
+	}
+	return databases, nil
+}
+
+// getElastiCacheServerlessCaches fetches all ElastiCache Serverless caches.
+func getElastiCacheServerlessCaches(ctx context.Context, client ElastiCacheClient) ([]ectypes.ServerlessCache, error) {
+	var out []ectypes.ServerlessCache
+	pager := elasticache.NewDescribeServerlessCachesPaginator(client,
+		&elasticache.DescribeServerlessCachesInput{},
+		func(opts *elasticache.DescribeServerlessCachesPaginatorOptions) {
+			opts.StopOnDuplicateToken = true
+		},
+	)
+	for i := 0; i < maxAWSPages && pager.HasMorePages(); i++ {
```

Comment:
Nit: Do we need to provide some kind of error or warning if we exit this loop because we hit `maxAWSPages` ?